### ISSUE TITLE
Import/Export of GXS IDs (and possible others)

### ISSUE THREAD

**Issue by sehraf** [RetroShare/RetroShare#1769]
Currently GXS IDs are bound to the location, so moving to a new one also brings a new fresh GXS ID. Therefore channel/forum/xyz rights (e.g. publish or admin rights) and reputation will get lost.
Also having the ability to backup IDs would be nice, too (e.g. when loosing a RS installation due to hard disc failure).

Attention must be given when using the same GXS id on multiple locations as this **will** result in **unpredictable** behavior. (RS was designed to expect an ID to exists exactly once!)

Relevant:
- GXS ID
  - reputation
  - circles
- forums/channels
  - admin rights
  - publish rights

**Comment by sehraf** [comment]
This is a hackish approach for exporting GXS IDs.
https://github.com/sehraf/RetroShare/commit/2399b3cc61960bfa2ec303103791300f8d91d875

**Comment by defnax** [comment]
But set new Admin(s) would be useful too without i need to export/import id to get the rights back. 

* Add Admin(s)

**Comment by csoler** [comment]
Is this a v0.6.6 blocker? It needs quite a lot of dev and lots of testing, and it is not a bug either. I think the label is misplaced here.

**Comment by defnax** [comment]
by the way maybe the part for posted is wrong @sehraf ?

when i share the permission i doesnt get the admin rights on other rs client.
only it says i has publish rights but for posted links.
but that makes no sence to share publish rights, every one can post when subscribed.
Share admin rights will be more useful for posted.

![image](https://user-images.githubusercontent.com/9952056/74087959-64cb9b80-4a91-11ea-9cfa-c920da217105.png)



`
if (mGrpType == CHANNEL_KEY_SHARE)
    {
        if (!rsGxsChannels)
            return;
          
        ui->headerFrame->setHeaderImage(QPixmap(":/images/channels.png"));
        ui->headerFrame->setHeaderText(tr("Share channel publish permissions"));
        ui->sharekeyinfo_label->setText(tr("You can allow your friends to publish in your channel, or send the publish permissions to another Retroshare instance of yours. Select the friends which you want to be allowed to publish in this channel. Note: it is currently not possible to revoke channel publish permissions."));
    }
    else if(mGrpType == FORUM_KEY_SHARE)
    {
        
        ui->headerFrame->setHeaderImage(QPixmap(":/images/konversation64.png"));
        ui->headerFrame->setHeaderText(tr("Share forum admin permissions"));
        ui->sharekeyinfo_label->setText(tr("You can let your friends know about your forum by sharing it with them. Select the friends with which you want to share your forum."));

    }
    else if (mGrpType == POSTED_KEY_SHARE)
    {
        if (!rsPosted)
            return;
        
        ui->headerFrame->setHeaderImage(QPixmap(":/images/posted_64.png"));    
        ui->headerFrame->setHeaderText(tr("Share topic admin permissions"));
        ui->sharekeyinfo_label->setText(tr("You can allow your friends to edit the topic. Select them in the list below. Note: it is not possible to revoke Posted admin permissions."));

    }`

https://github.com/RetroShare/RetroShare/blob/master/retroshare-gui/src/gui/gxs/GxsGroupShareKey.cpp

**Comment by csoler** [comment]
yes, sharing publish rights for posted links is irrelevant. It should not be in the menu.

**Comment by defnax** [comment]
> Currently GXS IDs are bound to the location, so moving to a new one also brings a new fresh GXS ID. Therefore channel/forum/xyz rights (e.g. publish or admin rights) and reputation will get lost. Also having the ability to backup IDs would be nice, too (e.g. when loosing a RS installation due to hard disc failure).
> 
> Attention must be given when using the same GXS id on multiple locations as this **will** result in **unpredictable** behavior. (RS was designed to expect an ID to exists exactly once!)
> 
> Relevant:
> 
> * GXS ID
>   
>   * reputation
>   * circles
> * forums/channels
>   
>   * admin rights
>   * publish rights

this not more works? @sehraf ? it crashes
